models/patient_readmission/app.py

```diff
@@ -15,75 +15,6 @@
 app = SimpleModelWrapper(model=model, encoder=encoder)
 app.run(production=True)
 
-from certifai.model.sdk import SimpleModelWrapper
-
-# All user-defined code that is to be used in the prediction wrapper must be in
-# instance methods in the wrapper class, or in imported modules
-# defined in set_global_imports
-
-class PythonModelWrapper(SimpleModelWrapper):
-    def __init__(self, *args, **kwargs):
-        SimpleModelWrapper.__init__(self, *args, **kwargs)
-
-    def set_global_imports(self):
-        """
-        Override this method to make external global imports
-        When using external imports override the method to provide necessary imports.
-        Make sure to mark them `global` to
-        be used by certifai interpreter correctly. set once,use throughout.
-        :return: None
-        """
-        global CleanPipeline
-        try:
-            from clean_pipeline import CleanPipeline
-        except ImportError:
-            pass
-
-    def soft_predict(self, npinstances):
-        """
-        Override this method for custom soft scoring model predictions
-        :param npinstances: np.ndarray
-        :return: np.ndarray
-        """
-        # reference model by using `self.model` e.g self.model.predict_proba(npinstances)
-        # default implementation for sklearn models below.
-
-        if not self._soft_scoring_validated:
-            self._validate_soft_scoring()
-        return self.model.predict_proba(npinstances)
-
-    def predict(self, npinstances):
-        """
-        Calls predict on the python model. Works out of the box with sklearn
-        models. Override this method for other model frameworks, or to map
-        outputs to desired class labels.
-        :param npinstances: np.ndarray
-        :return: np.ndarray
-        """
-        # reference model by using `self.model` e.g self.model.predict(npinstances)
-        return self.model.predict(npinstances)
-
-
-# These imports are used in launching the prediction service. They are not
-# used within the prediction service
-import os
-import pickle
-from utils import fetch_model, load_metadata
-
-def main():
-    with open('readmission_mlp.pkl', 'rb') as f:
-        saved = pickle.load(f)
-        model = saved.get('model')
-        encoder = saved.get('encoder', None)
-
-    app = PythonModelWrapper(model=model,
-                             encoder=encoder,
-                             host='0.0.0.0',
-                             supports_soft_scores=False
-                             )
-    # Production mode requires Certifai 1.3.6 or higher
-    app.run(production=True, log_level='warning', num_workers=3)
-
 
 if __name__ == '__main__':
     main()
```

Comment:
There is no longer a `main` function... I'm really not sure if the app setup should be in within the 'main' block or not. I don't think it matters unless we're using gunicorn or something similar...

Reply:
I removed the main, but we are using gunicorn in production mode. I do have a qualm vs the original main (see import of CleanPipeline) - I'm remembering now about having to jump through hoops to get imports used in the pickle to work. But it runs ok so now I'm wondering whether later gunicorn has fixed it or what?

Reply:
I don't remember/know what those problems were.. There are subtleties like class definitions can't be pickled, so the definition needs to be available when un-pickling.

Reason why gunicorn came to mind is that the workers are not running in the main thread (so gotta watch what code is/isn't called by main) - but I think that doesn't matter here because the gunicorn workers shouldn't import this module

Reply:
The model's running in the workers, so would need CleanPipeline, though.